app/models/mail_notification.rb

```diff
@@ -31,10 +31,16 @@ def self.[](name)
     self.find_by_name(name)
   end
 
-  def deliver(options)
-    mailer.constantize.send(method, options).deliver
-  rescue => e
-    logger.warn "Failed to send email notification #{name}: #{e}"
-    logger.debug e.backtrace
+  def deliver(*args)
+    # args can be anything really, treat it carefully
+    # handle args=[.., :users => [..]] specially and instantiate a single mailer per user, with :user set on each
+    if args.last.is_a?(Hash) && args.last.has_key?(:users)
+      options = args.pop
+      options.delete(:users).each do |user|
```

Comment:
Ah, for async tasks, it'd be nice if we could do the User lookup here by passing an array of ID's.   Trying to think about how I'll implement this in Katello, it might look something like this:

  https://github.com/Katello/katello/commit/b38e9eabdb961639d150e7f545610d0d8d45dcac

Not very dry for every async task having to do the lookup themselves, I think.